vm/src/builtins/pystr.rs

```diff
@@ -1197,6 +1242,38 @@ impl FindArgs {
     }
 }
 
+// Common reducer for forward and reverse str iterators.
+fn str_reduce(
+    list: PyRef<PyStr>,
+    position: Option<usize>,
+    reverse: bool,
+    vm: &VirtualMachine,
+) -> PyResult {
+    let attr = if reverse { "reversed" } else { "iter" };
```

Comment:
plus, no need to make this distinction any more.

Reply:
I agree with it. It's my miss. I sent v3 code. thanks!